apps/web/app/(roles)/vaga/[id]/RolePage.tsx

```diff
@@ -0,0 +1,285 @@
+        }
+        currentLine += sentence
+      })
+
+      if (currentLine) {
+        result.push(currentLine.trim())
+      }
+
+      return result
+    }
+
+    const formattedDescription = insertLineBreaks(description, 300)
+
+    return formattedDescription.map((line, index) => (
+      <p key={index} className="mb-4 whitespace-pre-line">
+        {line}
+      </p>
+    ))
+  }, [])
+
+  const isHtml = useCallback((text) => /<\/?[a-z][\s\S]*>/i.test(text), [])
+
+  const formatDate = useCallback((dateString, language) => {
+    const months = {
+      en: [
+        'Jan',
+        'Feb',
+        'Mar',
+        'Apr',
+        'May',
+        'Jun',
+        'Jul',
+        'Aug',
+        'Sep',
+        'Oct',
+        'Nov',
+        'Dec',
+      ],
+      pt: [
+        'Jan',
+        'Fev',
+        'Mar',
+        'Abr',
+        'Mai',
+        'Jun',
+        'Jul',
+        'Ago',
+        'Set',
+        'Out',
+        'Nov',
+        'Dez',
+      ],
+    }
+
+    const date = new Date(dateString)
+    const month = months[language === 'English' ? 'en' : 'pt'][date.getMonth()]
+    const day = date.getDate()
+    const year = date.getFullYear()
+
+    return `${day} ${month}, ${year}`
+  }, [])
+
+  const handleApply = useCallback(() => {
+    window.open(
+      `https://router.trampardecasa.com.br/api/role-access?roleId=${vaga.id}`,
+      '_blank'
+    )
+  }, [vaga.id])
+
+  const handleCopyLink = useCallback(() => {
+    navigator.clipboard.writeText(window.location.href)
+    setIsCopied(true)
+
+    setTimeout(() => {
+      setIsCopied(false)
+    }, COPY_TIMEOUT)
+  }, [])
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (
+        shareMenuRef.current &&
+        !shareMenuRef.current.contains(event.target)
+      ) {
+        setShowShareMenu(false)
+      }
+    }
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => document.removeEventListener('mousedown', handleClickOutside)
+  }, [])
+
+  const jobDetails = useMemo(
+    () => [
+      {
+        icon: <Laptop className="text-indigo-600" />,
+        label: isEnglish ? 'Work Type' : 'Tipo de Trabalho',
+        value: isEnglish ? 'Remote' : 'Remoto',
+      },
+      {
+        icon: <MapPin className="text-indigo-600" />,
+        label: isEnglish ? 'Location' : 'Localização',
+        value: vaga.country,
+      },
+      {
+        icon: <AlarmClock className="text-indigo-600" />,
+        label: isEnglish ? 'Job Type' : 'Tipo de Contrato',
+        value: 'Full-time',
+      },
+      {
+        icon: <DollarSign className="text-indigo-600" />,
+        label: isEnglish ? 'Salary' : 'Salário',
+        value: vaga.salary || (isEnglish ? 'Not informed' : 'Não informado'),
+      },
+      {
+        icon: <Calendar className="text-indigo-600" />,
+        label: isEnglish ? 'Posted On' : 'Publicado em',
+        value: formatDate(vaga.updatedAt, vaga.language),
+      },
+    ],
```

Comment:
This code is not performatic, for each interation, it is doing 6 checks of `isEnglish` variable.

While this may not be a performance problem in this application, it's not a good idea to write code this way, as it compounds over the time.

Please refactor to validate only a single time (returning different objects).